CMakeExternals.cmake

```diff
@@ -135,52 +140,54 @@ function(require_external NAME)
       return()
     endif()
 
-    set(BHTSNE_LIB_DEBUG "lib/${CMAKE_STATIC_LIBRARY_PREFIX}bhtsned${CMAKE_STATIC_LIBRARY_SUFFIX}")
-    set(BHTSNE_LIB_RELEASE "lib/${CMAKE_STATIC_LIBRARY_PREFIX}bhtsne${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    if(WIN32)
+      set(BHTSNE_LIB "lib/bhtsne.lib")
+    else()
+      set(BHTSNE_LIB "lib/libbhtsne.a")
+    endif()
 
-    add_external_project(bhtsne
+    add_external_project(bhtsne STATIC
       GIT_REPOSITORY https://github.com/lvdmaaten/bhtsne.git
       GIT_TAG "36b169c88250d0afe51828448dfdeeaa508f13bc"
-      BUILD_BYPRODUCTS "<INSTALL_DIR>/lib/${CMAKE_STATIC_LIBRARY_PREFIX}bhtsne<SUFFIX>${CMAKE_STATIC_LIBRARY_SUFFIX}"
-      DEBUG_SUFFIX d
+      BUILD_BYPRODUCTS "<INSTALL_DIR>/${BHTSNE_LIB}"
       PATCH_COMMAND ${CMAKE_COMMAND} -E copy
         "${CMAKE_SOURCE_DIR}/cmake/bhtsne/CMakeLists.txt"
         "<SOURCE_DIR>/CMakeLists.txt")
 
-    add_external_library(bhtsne STATIC
-      LIBRARY_DEBUG ${BHTSNE_LIB_DEBUG}
-      LIBRARY_RELEASE ${BHTSNE_LIB_RELEASE})
+    add_external_library(bhtsne
+      LIBRARY ${BHTSNE_LIB})
 
   # glfw3
   elseif(NAME STREQUAL "glfw3")
     if(TARGET glfw3)
       return()
     endif()
 
-    if (MSVC)
-      set(GLFW_PRODUCT "lib/glfw3dll.lib")
-      set(GLFW_IMPORT_LIBRARY "lib/glfw3dll.lib")
-      set(GLFW_LIBRARY "lib/glfw3.dll")
+    if(WIN32)
+      set(GLFW_IMPORT_LIB "lib/glfw3dll.lib")
+      set(GLFW_LIB "lib/glfw3.dll")
+      set(MOVE_CMD COMMANDS COMMAND ${CMAKE_COMMAND} -E copy "\"<INSTALL_DIR>/lib/glfw3.dll\" \"<INSTALL_DIR>/bin/glfw3.dll\""
+                            COMMAND ${CMAKE_COMMAND} -E remove -f \"<INSTALL_DIR>/lib/glfw3.dll\")
```

Comment:
The glfw dll is installed into the lib directory instead of the dll directory. Thus, not copying it to the correct folder, the MegaMol executables are not able to find it. Therefore, this has to be done at some point.